src/Material/Textfield.elm

```diff
@@ -349,8 +351,19 @@ view lift model options =
         Nothing -> []
 
 
-    listeners =
-      config.listeners
+    -- Internal events that require special handling because user provided events
+    -- might overlap with these
+    internal =
+      [[ ("focus", Decoder.succeed (lift Focus))
+       , ("blur", Decoder.succeed (lift Blur))
+       ]
+      , (case config.value of
```

Comment:
I know this was my invention originally, but I get confused every time I see it. Could we leave a comment saying why there's an `input` handler iff we're given a value?